### PR TITLE
fix progress bar validation batches

### LINKED ISSUE #7445: Progress bar shows twice the correct number of batches
## 🐛 Bug

For a mini dataset of 100 examples with batch_size=1 (dataset.__len__() returns 100), the progress bar shows 100.

I can't think of any reason why it's doubling in this manner.  I've tried different value for `val_check_interval`, `batch_size`, cpu and single / multiple GPU, and the problem persists.  Thanks for the help -- this isn't getting in the way of anything (Lightning is really helpful!) -- it's just a feng shui issue for me!

## Main Training Script
```python
    args = parser.parse_args()
    data_dir = os.path.join(NYP_DIR, 'planning')    
    cui_vocab = get_vocab()
    cui_V = len(cui_vocab)
    model = EntityPlanner(args, cui_V + 5)
    data_module = EntityPlanningDataModule(data_dir, cui_vocab=cui_vocab, debug=args.debug)
    weight_dir = os.path.expanduser('~/weights/planning')
    os.makedirs(weight_dir, exist_ok=True)
    experiment_dir = os.path.join(weight_dir, args.experiment)
    os.makedirs(os.path.join(experiment_dir, 'wandb'), exist_ok=True)  # Only way to make sure it's writable

    logger = pl_loggers.WandbLogger(
        name=args.experiment,
        save_dir=experiment_dir,
        offline=args.debug or args.offline,
        project='xxx',
        entity='xxx',
    )

    num_gpus = torch.cuda.device_count() if torch.cuda.is_available() and not args.cpu else None
    if num_gpus is not None and args.debug:
        num_gpus = 1
    print('Num GPUs --> {}'.format(num_gpus))
    pin_memory = num_gpus is not None and not args.debug
    precision = 16 if num_gpus is not None else 32

    checkpoint_callback = ModelCheckpoint(
        monitor='val_loss',
        save_top_k=1,
        save_last=True,
        mode='min'
    )

    early_stopping = EarlyStopping('val_correlation', patience=3)
    callbacks = [early_stopping, checkpoint_callback]

    accumulate_grad_batches = args.batch_size if num_gpus is None or num_gpus == 1 else args.batch_size // num_gpus

    trainer = pl.Trainer.from_argparse_args(
        args,
        logger=logger,
        precision=precision,
        accelerator=None if num_gpus is None or num_gpus == 1 else 'ddp',
        gpus=num_gpus,
        terminate_on_nan=True,
        default_root_dir=experiment_dir,
        gradient_clip_val=0.1,
        val_check_interval=0.0 if args.debug else 0.2,
        num_sanity_val_steps=0 if args.debug else 2,
        accumulate_grad_batches=accumulate_grad_batches
    )

    if args.auto_lr_find:
        trainer.tune(model, data_module)
    else:
        print('Starting training...')
        trainer.fit(model, data_module)
        print(f'Best weights saved --> {checkpoint_callback.best_model_path}')`
```

### PR DESCRIPTION
**Pull request by edgarriba**
## What does this PR do?

Fixes #7445

TODO:
- [ ] tests

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [ ] Did you make sure to update the documentation with your changes? (if necessary)
- [ ] Did you write any new necessary tests? (not for typos and docs)
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [ ] Is this pull request ready for review? (if not, please submit in draft mode)
 - [ ] Check that all items from **Before submitting** are resolved
 - [ ] Make sure the title is self-explanatory and the description concisely explains the PR
 - [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by tchaton** [comment]
Hey @edgarriba,

Is it ready for review ?

Best,
T.C

**Comment by edgarriba** [comment]
@tchaton waiting for a response in the original thread #7445 to make sure this makes sense.
